integration_tests/commands/websocket/set_test.go

```diff
@@ -0,0 +1,262 @@
+package websocket
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	testifyAssert "github.com/stretchr/testify/assert"
+
+	"gotest.tools/v3/assert"
+)
+
+type TestCase struct {
+	name     string
+	commands []WebsocketCommand
+	expected []interface{}
+}
+
+func TestSet(t *testing.T) {
```

Comment:
Instead of writing new tests for WS, I see that the input and output are the same in Async and WS.
I'd see how we can reuse the same async tests even for websockets with minimal rewrite, i.e just have the different fire commands?

Reply:
Here's what needs to change -

We don't need `WebsocketCommand`.

Change FireCommand to -
- func (e *WebsocketCommandExecutor) FireCommand(conn *websocket.Conn, cmd string) (interface{}, error) 
- Copy tests as is from async folder -

```

type TestCase struct {
	name     string
	commands []string
	expected []interface{}
}

func TestSet(t *testing.T) {
	exec := NewWebsocketCommandExecutor()

	testCases := []TestCase{
		{
			name:     "Set and Get Simple Value",
			commands: []string{"SET k v", "GET k"},
			expected: []interface{}{"OK", "v"},
		},
		{
			name:     "Set and Get Integer Value",
			commands: []string{"SET k 123456789", "GET k"},
			expected: []interface{}{"OK", float64(123456789)},
		},
		{
			name:     "Overwrite Existing Key",
			commands: []string{"SET k v1", "SET k 5", "GET k"},
			expected: []interface{}{"OK", "OK", float64(5)},
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			conn := exec.ConnectToServer()
			// deleteTestKeys([]string{"k"}, store)
			_, err := exec.FireCommand(conn, "DEL k")
			testifyAssert.NoError(t, err)

			for i, cmd := range tc.commands {
				result, err := exec.FireCommand(conn, cmd)
				testifyAssert.NoError(t, err)
				assert.DeepEqual(t, tc.expected[i], result)
			}
		})
	}
}
```

3) Fix small things like data type etc. With this, we can be sure that all commands in WS work the same way as in RESP.
4) Get all tests ported, unless you see challenges for some, in which case, let us know